src/toxcore/io_tokio.rs

```diff
@@ -32,3 +32,51 @@ pub fn send_all_to<T: Send + 'static, S, Tx, E: Debug>(tx: &Tx, s: S) -> impl Fu
         .map(|_tx| ()) // ignore tx because it was cloned
 }
 
+/// `Sink` type that can either be empty or store inner `Sink`. When it's empty
+/// it drops all sent elements.
+#[derive(Debug, Clone)]
```

Comment:
derive Default,

pub fn new() will be: { Default::default() }

Reply:
Nope, it won't work. `derive` requires trait implementation for all generic type parameters. In our case it will require `Default` for senders which they don't implement.

Reply:
Did you try?
https://doc.rust-lang.org/src/core/option.rs.html#1042-1046

Reply:
Looks like a bug. I will file it to rustlang.

Reply:
https://stackoverflow.com/questions/46079916/possible-bug-in-generic-implementation-of-default